Index: rubyfull-3.1.1/ruby-build/ruby/build_dir/ruby-3.1.1/configure.ac
===================================================================
--- a/configure.ac
+++ b/configure.ac
@@ -3395,7 +3395,7 @@ AC_ARG_WITH(soname,
 LIBRUBY_LDSHARED=${DLDSHARED=${LDSHARED}}
 LIBRUBY_DLDFLAGS=$DLDFLAGS
 LIBRUBY_SO='lib$(RUBY_SO_NAME).$(SOEXT).$(RUBY_PROGRAM_VERSION)'
-LIBRUBY_SONAME='lib$(RUBY_SO_NAME).$(SOEXT).$(RUBY_API_VERSION)'
+LIBRUBY_SONAME='lib$(RUBY_SO_NAME).$(SOEXT)'
 LIBRUBY_ALIASES='lib$(RUBY_SO_NAME).$(SOEXT)'
 ENABLE_SHARED=no

